ArduPlane/mode_guided.cpp

```diff
@@ -29,24 +68,62 @@ bool ModeGuided::_enter()
 
 void ModeGuided::update()
 {
+    switch (submode) {
+        case SubMode::Guided: {
 #if HAL_QUADPLANE_ENABLED
-    if (plane.auto_state.vtol_loiter && plane.quadplane.available()) {
-        plane.quadplane.guided_update();
-        return;
-    }
+            if (plane.auto_state.vtol_loiter && plane.quadplane.available()) {
+                plane.quadplane.guided_update();
+                return;
+            }
 #endif
-    plane.calc_nav_roll();
-    plane.calc_nav_pitch();
-    plane.calc_throttle();
+            plane.calc_nav_roll();
+            plane.calc_nav_pitch();
+            plane.calc_throttle();
+            break;
+        }
+
+#if AP_SCRIPTING_ENABLED
+        case SubMode::NAV_Script_time: {
+            // Reset desired angles
+            plane.nav_roll_cd = plane.ahrs.roll_sensor;
+            plane.nav_pitch_cd = plane.ahrs.pitch_sensor;
+
+            // Check for timeout
+            plane.nav_scripting_check_timeout();
+
+            // Reset to previous mode if command done
+            if (!plane.nav_scripting_active()) {
+                plane.set_mode(*plane.previous_mode, ModeReason::NAV_SCRIPTING_COMPLETE);
+                // Should probably check for failure here, or add ability to force mode change
```

Comment:
None of the modes that allow tricks ever actually fail to enter, but we should probably deal with this properly anyway. We could have "trickle down fail safes" ie RTL then loiter if RTL fails then land if loiter fails. Or we could add the ability to force a mode change.